docs/source/autograd.rst

```diff
@@ -53,6 +53,55 @@ Locally disabling gradient computation
 
 .. autoclass:: set_grad_enabled
 
+.. _default-grad-layouts:
+
+Default gradient layouts
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+When a non-sparse ``param`` receives a non-sparse gradient during
+:func:`torch.autograd.backward` or :func:`torch.Tensor.backward`
+``param.grad`` is accumulated as follows.
+
+If ``param.grad`` is initially ``None``:
+
+1. If ``param``'s memory is non-overlapping and dense, ``.grad`` is
+   created with strides matching ``param``.
+2. Otherwise, ``.grad`` is created with rowmajor-contiguous strides.
+
+If ``param`` already has a non-sparse ``.grad`` attribute:
+
+3. If ``create_graph=False``, ``backward()`` accumulates into ``.grad``
+   in-place, which preserves its strides.
+4. If ``create_graph=True``, ``backward()`` replaces ``.grad`` with a
+   new accumulated tensor, and attempts (but does not guarantee) matching
+   the preexisting ``.grad``'s strides.
+
+The default behavior (letting ``.grad``\ s be ``None`` before the first
+``backward()``, such that their layout is created according to 1. or 2.,
+and retained over time according to 3. or 4) is recommended for best performance.
+Calls to ``model`` or ``optimizer.zero_grad()`` will not affect ``.grad``
```

Comment:
nit: one here has parenthesis and not the other. Or are you missing the `.zero_grad()` for the model? Same below.

Reply:
when i wrote it my thought was `model.zero_grad()` was implied, but that's not clear, i'll add it